ibm_vpd_utils.hpp

```diff
@@ -77,6 +77,68 @@ string encodeKeyword(const string& kw, const string& encoding);
 string readBusProperty(const string& obj, const string& inf,
                        const string& prop);
 
+/** @brief A templated function to read D-Bus properties.
+ *
+ *  @param[in] service - Service path
+ *  @param[in] object - object path
+ *  @param[in] inf - interface
+ *  @param[in] prop - property whose value is fetched
+ *  @return The property value of its own type.
+ */
+template <typename T>
+T readDBusProperty(const string& service, const string& object,
```

Comment:
Not sure why we have duplicate function for this.

Reply:
As discussed before these are cherry-pick change, and planned not to change for 1020.